### PR TITLE
Add support for encoding lossy webp images with alpha channels

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This PR adds support for encoding lossy webp images with alpha channels.
The alpha data can be either compressed (with webp lossless compression) or uncompressed.

Related to #1802

### CODE REVIEW

File: src/ImageSharp/Formats/Webp/AlphaEncoder.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        /// <param name="memoryAllocator">The memory manager.</param>
+        /// <param name="compress">Indicates, if the data should be compressed with the lossless webp compression.</param>
+        /// <param name="size">The size in bytes of the alpha data.</param>
+        /// <returns>The encoded alpha data.</returns>
+        public static IMemoryOwner<byte> EncodeAlpha<TPixel>(Image<TPixel> image, Configuration configuration, MemoryAllocator memoryAllocator, bool compress, out int size)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            int width = image.Width;
+            int height = image.Height;
+            IMemoryOwner<byte> alphaData = ExtractAlphaChannel(image, configuration, memoryAllocator);
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Webp/AlphaEncoder.cs:34]: I can't see where this is getting disposed?

Comment (brianpopow) [reply src/ImageSharp/Formats/Webp/AlphaEncoder.cs:34]: Its done in the `Dispose` of the `Vp8Encoder`

Comment (brianpopow) [reply src/ImageSharp/Formats/Webp/AlphaEncoder.cs:34]: Maybe it would be better to keep the responsibility of disposing to the `AlphaEncoder` instead?

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Webp/AlphaEncoder.cs:34]: Yeah, I think that's wise. Difficult to keep track of then.

File: src/ImageSharp/Formats/Webp/Lossy/Vp8Encoder.cs [hunk 2]
```diff
@@ -71,10 +71,7 @@ internal class Vp8Encoder : IDisposable
         /// </summary>
         private int uvAlpha;
 
-        /// <summary>
-        /// Scratch buffer to reduce allocations.
-        /// </summary>
-        private readonly int[] scratch = new int[16];
+        private readonly bool alphaCompression;
 
         private readonly byte[] averageBytesPerMb = { 50, 24, 16, 9, 7, 5, 3, 2 };
```
Comment (gfoidl) [inline src/ImageSharp/Formats/Webp/Lossy/Vp8Encoder.cs:76]:
```suggestion
        // This uses C#'s optimization to refer to the static data segment of the assembly, no allocation occurs.
        private static ReadOnlySpan<byte> AverageBytesPerMb => new byte[] { 50, 24, 16, 9, 7, 5, 3, 2 };
```
While you're on it.

Need to update https://github.com/SixLabors/ImageSharp/blob/484bd77d8d2e0cca5bdbad6e528337fc21b4af94/src/ImageSharp/Formats/Webp/Lossy/Vp8Encoder.cs#L327 to new name too.

**Review by JimBobSquarePants — APPROVED** [review]
LGTM. Thanks for pushing this through so quickly!

**Pull request merged**